sport-moment-nft/contracts/EPL.cdc

```diff
@@ -0,0 +1,1124 @@
+    ///
+    pub fun getEditionData(id: UInt64): EditionData? {
+        if EPL.editionByID[id] == nil{
+            return nil
+        }
+
+        return EPL.EditionData(id: id)
+    }
+
+    // Validate tags exist
+    //
+    pub fun validateTags(tagIds: [UInt64]): Bool {
+        for tagId in tagIds {
+            if EPL.tagByID[tagId] == nil {
+                return false
+            }
+        }
+        return true
+    }
+
+    //------------------------------------------------------------
+    // NFT
+    //------------------------------------------------------------
+
+    /// A Moment NFT
+    ///
+    pub resource NFT: NonFungibleToken.INFT, MetadataViews.Resolver {
+        pub let id: UInt64
+        pub let editionID: UInt64
+        pub let serialNumber: UInt64
+        pub let mintingDate: UFix64
+
+        /// Destructor
+        ///
+        destroy() {
+            emit MomentNFTBurned(id: self.id, editionID: self.editionID, serialNumber: self.serialNumber)
+        }
+
+        /// NFT initializer
+        ///
+        init(
+            editionID: UInt64,
+            serialNumber: UInt64
+        ) {
+            pre {
+                EPL.editionByID[editionID] != nil: "no such editionID"
+                EditionData(id: editionID).maxEditionMintSizeReached() != true: "max edition size already reached"
+            }
+
+            self.id = self.uuid
+            self.editionID = editionID
+            self.serialNumber = serialNumber
+            self.mintingDate = getCurrentBlock().timestamp
+
+            emit MomentNFTMinted(id: self.id, editionID: self.editionID, serialNumber: self.serialNumber)
+        }
+
+        pub fun assetPath(): String {
+            let editionData = EPL.getEditionData(id: self.editionID)!
+            let playDataID: String = EPL.PlayData(id: editionData.playID).metadata["PlayDataID"] ?? ""
+            return "https://assets.eplonflow.com/editions/".concat(playDataID).concat("/play_").concat(playDataID)
+        }
+
+        pub fun getImage(imageType: String): String {
+            return self.assetPath().concat("__").concat(imageType).concat("_2880_2880_").concat(".png")
+        }
+
+        pub fun getVideo(videoType: String): String {
+            return self.assetPath().concat("__").concat(videoType).concat("_1080_1080_").concat(".mp4")
+        }
+
+        /// get the name of an nft
+        ///
+        pub fun name(): String {
+            let editionData = EPL.getEditionData(id: self.editionID)!
+            let playerKnownName: String = EPL.PlayData(id: editionData.playID).metadata["PlayerKnownName"] ?? ""
+            let playerFirstName: String = EPL.PlayData(id: editionData.playID).metadata["PlayerFirstName"] ?? ""
+            let playerLastName: String = EPL.PlayData(id: editionData.playID).metadata["PlayerLastName"] ?? ""
+            let playType: String = EPL.PlayData(id: editionData.playID).metadata["PlayType"] ?? ""
```

Comment:
We've gotten some complaints from the community that the key names for metadata in studio projects are not human readable, so we should try to make that more readable from now on by using correct spacing and such. For example, `PlayerKnownName` should be changed to `Player Known Name`

Reply:
Done